src/theme/BlogPostItem/index.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from "react";
+import BlogPostItem from "@theme-original/BlogPostItem";
+import type BlogPostItemType from "@theme/BlogPostItem";
+import type { WrapperProps } from "@docusaurus/types";
+import styles from "@docusaurus/theme-classic/src/theme/Heading/styles.module.css";
+
+import { useBlogPost } from "@docusaurus/plugin-content-blog/client";
+import { useColorMode } from "@docusaurus/theme-common";
+import Giscus from "@giscus/react";
+
+type Props = WrapperProps<typeof BlogPostItemType>;
+
+// wrapper for the BlogPostItem component to show Giscus discussions below blog posts
+// the initial implementation has been generated by this command:
+//   yarn swizzle @docusaurus/theme-classic BlogPostItem -- --wrap
+// @see https://docusaurus.io/docs/swizzling#wrapping
+export default function BlogPostItemWrapper(props: Props): JSX.Element {
+  const frontMatter = useBlogPost().metadata.frontMatter;
+  // allow hiding the discussion via the front matter config,
+  // use "discussion: false" to disable discussion in a specific post
+  const discussionEnabled = frontMatter.discussion !== false;
+
+  // if discussion is disabled then just return the standard blog page content
+  if (!discussionEnabled) return <BlogPostItem {...props} />;
+
+  const discussionId =
+    frontMatter.discussion_id && Number(frontMatter.discussion_id);
+
+  const [discussionUrl, setDiscussionUrl] = useState(
+    discussionId > 0
+      ? `https://github.com/agama-project/agama-project.github.io/discussions/${discussionId}`
+      : ""
+  );
+
+  // handle metadata messages from the Giscus iframe
+  function handleGiscusMessage(event: MessageEvent) {
+    // check the origin
+    if (
+      event.origin !== "https://giscus.app" ||
+      event.source !==
+        document
+          .querySelector("giscus-widget")
+          .shadowRoot.querySelector("iframe").contentWindow
+    )
+      return;
+
+    // check the data
+    if (!(typeof event.data === "object" && event.data.giscus)) return;
+
+    const giscusData = event.data.giscus;
+    if (
+      typeof giscusData.discussion === "object" &&
+      giscusData.discussion.url
+    ) {
+      setDiscussionUrl(giscusData.discussion.url);
+    }
+  }
+
+  // register/unregister the Giscus message handler
+  useEffect(() => {
+    window.addEventListener("message", handleGiscusMessage);
+    return () => {
+      window.removeEventListener("message", handleGiscusMessage);
+    };
+  }, []);
+
+  // if there is an already existing discussion show a link to it
+  const discussionLink = discussionUrl ? (
```

Comment:
I think we can drop this and make the whole thing a bit short. Rationale: we are already stating that alternatively user can go to Agama GitHub project page

Reply:
I mean. It's great to help users as much as possible and I see this contains a direct link to the discussion. But fue that it is already a bunch of text and the user can comment right away... If she/he decided to go to the repo, I assume they know where to look for  :)

Reply:
Let me explain it with an example now that I'm on front of the laptop instead on a mobile device :)

In its current shape, you have 

 > We are interested in your feedback, feel free to leave a comment here!
> 
> To comment here you need a [GitHub](https://github.com/) account and authorize the [Giscus](https://giscus.app/) application. Or you can comment directly at [this GitHub discussion](https://github.com/agama-project/agama-project.github.io/discussions/20) without authorizing the Giscus application.
> 
> Alternatively you can reach us at the [#yast](https://web.libera.chat/#yast) channel at the [Libera.Chat](https://libera.chat/), [YaST Development](https://lists.opensuse.org/archives/list/yast-devel@lists.opensuse.org) mailing list or the [Agama project](https://github.com/agama-project/agama) at GitHub.

and my proposal is

> We are interested in your feedback, feel free to leave a comment here!
> 
> :information_source:  To comment here you need a [GitHub](https://github.com/) account and authorize the [Giscus](https://giscus.app/) application.
> 
> Alternatively you can reach us at the [#yast](https://web.libera.chat/#yast) channel at the [Libera.Chat](https://libera.chat/), [YaST Development](https://lists.opensuse.org/archives/list/yast-devel@lists.opensuse.org) mailing list or the [Agama project](https://github.com/agama-project/agama) at GitHub.

Where the "Agama project" could be the link to the already started discussion if it exists.

But now I wrote it, I realize discussions are in agama-project/agama-project and not in agama-project/agama. But you got the idea.

Reply:
I think it is not a duplication. The discussion link point directly to the blog post discussion. The GitHub link is a generic contact, can be used for reporting issues, code contribution, etc...

I'd keep both.